.devcontainer/cmake/toolchains/arm-none-eabi-gcc.cmake

```diff
@@ -0,0 +1,47 @@
+#
+# Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+# Copyright Amazon.com Inc. or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+
+# armv7m cross-compilation
+#
+set(CMAKE_C_COMPILER arm-none-eabi-gcc CACHE FILEPATH "C compiler")
+set(CMAKE_CXX_COMPILER  arm-none-eabi-g++ CACHE FILEPATH "C++ compiler")
+set(CMAKE_ASM_COMPILER  arm-none-eabi-gcc CACHE FILEPATH "assembler")
+
+set(CMAKE_SYSTEM_NAME Generic)
+# This should be board-specific but we also should have this set by the toolchain. Need to fix then when we add new
+# target processors.
+set(CMAKE_SYSTEM_PROCESSOR "cortex-m7")
+
+set(TOOLCHAIN_PREFIX arm-none-eabi)
+set(CMAKE_TRY_COMPILE_TARGET_TYPE STATIC_LIBRARY)
+
+set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
+set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
+set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
+set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE ONLY)
+
+
+# This are compile options for all modules we compile. Architecture-specific stuff should be inherited from the
+# board library's compile options since it's that module that defines what CPU we're running on.
+add_compile_options(
+    -fdata-sections
+    -ffunction-sections
+    -fverbose-asm
+    -mcpu=cortex-m7
```

Comment:
Isn't this supposed to be set from `CMAKE_SYSTEM_PROCESSOR` by CMake?

Reply:
This doesn't seem to work based on my experimentation. The documentation for `CMAKE_SYSTEM_PROCESSOR` reads:

> When cross-compiling, a [CMAKE_TOOLCHAIN_FILE](https://cmake.org/cmake/help/latest/variable/CMAKE_TOOLCHAIN_FILE.html#variable:CMAKE_TOOLCHAIN_FILE) should set the CMAKE_SYSTEM_PROCESSOR variable to match target architecture that it specifies (via [CMAKE_\<LANG\>_COMPILER](https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER.html#variable:CMAKE_%3CLANG%3E_COMPILER) and perhaps [CMAKE_\<LANG\>_COMPILER_TARGET](https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER_TARGET.html#variable:CMAKE_%3CLANG%3E_COMPILER_TARGET)).

which suggests one still has to set the compiler flag manually.